components/kyma-environment-broker/internal/broker/instance_update.go

```diff
@@ -45,7 +48,10 @@ func (b *UpdateEndpoint) Update(ctx context.Context, instanceID string, details
 	logger.Infof("Update asyncAllowed: %v", asyncAllowed)
 
 	instance, err := b.instanceStorage.GetByID(instanceID)
-	if err != nil {
+	if err != nil && dberr.IsNotFound(err) {
+		logger.Errorf("unable to get instance: %s", err.Error())
+		return domain.UpdateServiceSpec{}, apiresponses.NewFailureResponse(err, http.StatusNotFound, fmt.Sprintf("could not execute update for instanceID %s", instanceID))
```

Comment:
I tried to take an inspiration from:
https://github.com/kyma-project/control-plane/blob/634f84ee59828dc72c6a4aae6c3632ab11464e9b/components/kyma-environment-broker/internal/broker/instance_deprovision.go#L56

but I think the if-else block reads a bit easier than the switch case.